src/D_functions_and_procedures.sql

```diff
@@ -362,4 +362,192 @@ $$ LANGUAGE plpgsql;
 
 -- D12 TEST
 SELECT * FROM best_match_titles(ARRAY['apple', 'mads', 'mikkelsen']);
-SELECT title FROM release WHERE media_id=47460;
\ No newline at end of file
+SELECT title FROM release WHERE media_id=47460;
+
+
+
+-------------------D9 SIMILAR MOVIES SEARCH-------------------------
+--------------------------------------------------------------------
+
+--FINAL FUNCTION SIMILAR MOVIES
+--THIS COMBINES OTHER FUNCTIONS WHICH ARE DECLARED LOWER IN THIS FILE
+CREATE OR REPLACE FUNCTION get_similar_movies(input_media_id INTEGER)
+RETURNS TABLE (
+	media_id INTEGER,
+	count INTEGER
+) AS $$
+
+DECLARE
+	input_media RECORD;
+BEGIN
+
+  --get title id and country of input_media_id
+	SELECT DISTINCT m.media_id, r.title, mpc.country_id INTO input_media
+	FROM media m
+	JOIN release r USING (media_id)
+	LEFT JOIN media_production_country mpc USING(media_id)
+	WHERE m.media_id = input_media_id;
+	
+	RETURN query
+	SELECT com_res.media_id, SUM(com_res.count)::INTEGER AS total_count
+	FROM (
+		SELECT *  FROM get_count_of_movies_with_same_actors(input_media_id) 
+		UNION ALL
+		SELECT *  FROM find_movies_from_the_same_country(input_media.country_id)
+		UNION ALL
+		SELECT *  FROM find_movie_similar_titles(input_media.title,3)
+		UNION ALL
+		SELECT *  FROM find_movies_with_similar_crew(input_media_id)
+	) AS com_res
+	GROUP BY com_res.media_id
+	ORDER BY total_count DESC;
+END;
+$$ LANGUAGE plpgsql;
+
+-------------------------------------------------------
+---------get_count_of_movies_with_same_actors----------
+	
+CREATE OR REPLACE FUNCTION get_count_of_movies_with_same_actors(input_media_id INTEGER)
+RETURNS TABLE(
```

Comment:
Could you maybe also return the title of the media, this is for now done by joining the release table and finding the title that has the type original?


This is regarding all the functions

Reply:
done